files/borgback-adfc-final.sh

```diff
@@ -0,0 +1,118 @@
+BORG_USER="borguser"                                    # Benutzerkonto auf Backup-Client / FIX! NICHT ÄNDERN
+
+PRUNE_DAYS="7"                                          # Anzahl der vorzuhaltenden Tagessicherungen
+PRUNE_WEEKS="4"                                         # Anzahl der vorzuhaltenden Wochensicherungen
+PRUNE_MONTHS="12"                                       # Anzahl der vorzuhaltenden Monatssicherungen
+PRUNE_YEARS="2"                                         # Anzahl der vorzuhaltenden Jahressicherungen
+
+MAIL_SENDER="absender@e-mail.tld"                       # Absender
+MAIL_RECEIVER="empfaenger@e-mail.tld"                   # Empfänger
+MAIL_PASSWD="smtp-passwort"                             # SMTP-Passwort
+MAIL_SERVER="mailserver.hoster.tld"                     # SMTP-Server
+MAIL_SUBJECT="ADFC Backup-Report"                       # Betreffzeile / sollte so bleiben
+MAIL_BODY="Freier Text für E-Mail-Body."                # Body Text
+MAIL_ATTACHM="anhang1.ext /pfad/zum/anhang2.ext"        # weitere Anhänge hier eintragen (Backup-Logdatei geht standardmäßig immer mit raus)
+
+
+###########################################################################################
+############ AB HIER KEINE ANPASSUNGEN MEHR ERFORDERLICH! #################################
+###########################################################################################
+
+# Per Lock-Datei verhindern, dass das Programm doppelt gestartet wird
+exec 200>${LOCK_FILE} ||  exit 3
+flock -n 200
+
+# ---------- Einlesen der verbundenen Backup-Clients / Ermitteln der IP -------------------
+
+# Ist kein Client verbunden wird auch nichts unternommen
+if [ "$(ls -A $CLIENTS_ONLINE_DIR)" == "" ]; then
+   exit 1
+fi
+
+# Einlesen aller Verbindungen
+ALLCLIENTS=()
+for CLIENT in $CLIENTS_ONLINE_DIR/*; do
+    ALLCLIENTS[${#ALLCLIENTS[@]}]="$CLIENT"
+
+    # Alle in CLIENTS_ONLINE_DIR vorhandenen Verbindungen testen, wenn Offline löschen. Nur 192er-Netz zulässig
+    for CLIENT in "${ALLCLIENTS[@]}"; do
+        TEST_IP=$(echo "$CLIENT" | cut -d '=' -f 2)
+        if [ $(echo "$TEST_IP" | cut -d '.' -f 1) -ne 192 ]; then
+           rm $CLIENT
+        fi
+        ping -c 3 $TEST_IP 2>&1> /dev/null
+        RESULT=$?
+        if [ $RESULT == 0 ]; then
+           IPADDR=$TEST_IP
+           ONLINE_CLIENT=$CLIENT
+        else
+           rm $CLIENT
+        fi
+    done
+done
+
+# Falls keine Verbindung funktionierte ist das Verzeinis leer. Dann Abbruch!
+if [ "$(ls -A $CLIENTS_ONLINE_DIR)" == "" ]; then
+   exit 2
+fi
+
+
+
+# ---------- Backup-Routine ----------------------------------------------------------------
+
+# Repo-Init legt beim erstmaligen Aufruf ein Repository an. Weitere Aufrufe beenden sich ohne Funktion oder Fehler (Rückgabewert=2 spielt hier keine Rolle).
+BORG_PASSPHRASE="$BAK_SECRET" borg init --encryption=repokey --show-rc $BORG_USER@$IPADDR:$REPO_PATH/$REPO_NAME 2>&1
```

Comment:
borg init nicht in dieses Skript weil dann das mit den Forced-Commands nicht geht.